commands/FixturesController.php

```diff
@@ -0,0 +1,608 @@
+                $proposalContentHistory = new ProposalContentHistory();
+                $proposalContentHistory->proposal_id = $proposal->id;
+                $proposalContentHistory->date = Util::getDateTimeFormattedForDatabase(
+                    \DateTime::createFromFormat('U', random_int(
+                        $editionMinDate->format('U'),
+                        $editionMaxDate->format('U'),
+                    ))
+                );
+                $proposalContentHistory->content = self::getLorem(self::PROPOSAL_MIN_WORDS_COUNT, self::PROPOSAL_MAX_WORDS_COUNT);
+
+                if(!$proposalContentHistory->save()) {
+                    throw new CannotSaveException($proposalContentHistory);
+                }
+            }
+        }
+    }
+
+    /**
+     * Generates proposal file editions between the proposal submit date and now
+     * @throws CannotSaveException
+     */
+    private function generateProposalFileEditions()
+    {
+        foreach($this->proposals as $proposal) {
+            $editionMinDate = Util::getDateTimeFromDatabaseString($proposal->date);
+            $editionMaxDate = new \DateTime();
+
+            $editionsCount = random_int(self::MIN_FILE_EDITIONS_PER_PROPOSAL, self::MAX_FILE_EDITIONS_PER_PROPOSAL);
+
+            $this->println("Generating $editionsCount file editions for proposal {$proposal->id} of {$proposal->submitter->email}");
+
+            for($i = 0; $i < $editionsCount; ++$i) {
+                $proposalFileHistory = new ProposalFileHistory();
+                $proposalFileHistory->proposal_id = $proposal->id;
+                $proposalFileHistory->date = Util::getDateTimeFormattedForDatabase(
+                    \DateTime::createFromFormat('U', random_int(
+                        $editionMinDate->format('U'),
+                        $editionMaxDate->format('U'),
+                    ))
+                );
+                $proposalFileHistory->path = $proposal->id . '.jpg';
+
+                if(!$proposalFileHistory->save()) {
+                    throw new CannotSaveException($proposalFileHistory);
+                }
+            }
+        }
+    }
+
+    /**
+     * Generates proposal reviews between the proposal submit date and now
+     * @throws CannotSaveException
+     */
+    private function generateReviews()
+    {
+        foreach($this->proposals as $proposal) {
+            $reviewMinDate = Util::getDateTimeFromDatabaseString($proposal->date);
+            $reviewMaxDate = new \DateTime();
+
+            $reviewsCount = random_int(self::MIN_REVIEWS_PER_PROPOSAL, self::MAX_REVIEWS_PER_PROPOSAL);
+
+            $this->println("Generating $reviewsCount reviews for proposal {$proposal->id} of {$proposal->submitter->email}");
+
+            $reviewersWhoCanReview = $this->reviewers;
+            for($i = 0; $i < count($reviewersWhoCanReview); ++$i) {
+                if($reviewersWhoCanReview[$i]->id === $proposal->id) {
```

Comment:
It should be proposal->submitter_id no ?

Reply:
Of course, my bad